### ISSUE TITLE
Option to group subscriptions by Channel Groups in Subscriptions tab

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#3590]
#### Is your feature request related to a problem? Please describe it

Right now, there is no convenient way to look at what channels you have in your channel groups. Currently, you have to long press on a group, open the subscriptions picker and then see the ticked channels.

Another issue is that viewing and managing the list of subscriptions starts to become increasingly tedious by the time their number reaches the triple digits.

#### Describe the feature you want

There should be an option to group channels by their channel groups (hehe), replacing the simple list we see in the Subscriptions tab.

What I envision is that when you choose to group subscriptions this way, just like the arrow next to the Subscriptions header expands to show import export options, so should we see a list of channel groups which can be expanded to show their respective channels.

A final header (which could simply be "Ungrouped") could then show all the channels which don't belong to any channel group.

Whether or not it is better for them to be expanded by default, I don't know.

**Comment by opusforlife2** [comment]
P.S.: I'm aware that channels which belong to multiple groups will be duplicated. I don't see why that should matter.

**Comment by Stypox** [comment]
Wouldn't  #3559 be a suitable solution?

**Comment by opusforlife2** [comment]
> Wouldn't #3559 be a suitable solution?

It wouldn't solve the second problem I mentioned:

> Another issue is that viewing and managing the list of subscriptions starts to become increasingly tedious by the time their number reaches the triple digits.

Plus, making an existing screen easier to manage would be better than introducing yet another tab on a Feed Page. It will reduce redundancy instead of increasing it.

**Comment by opusforlife2** [comment]
Actually, come to think of it, with this implemented, you could also easily do away with the channel group boxes above the subscriptions list:

- As I said above, drop down arrows next to channel group headers could expand to show the contained subscriptions
- The Channel Group headers _themselves_ could be made tappable, opening their Feeds.

**Comment by MD77MD** [comment]
#3315
Older feed UI switch #3315

although the new feed UI has been a long awaited feature.. old feed UI definitely had its uses... combining both would be really awesome
I would suggest to add swipe left and right (like the one used between comments and related videos) to switch between two panels
-the first panel is used for listing feeds by upload time (new feed UI)
-the second panel will show latest video from each channel (old feed UI).. i would also like the ability added to go to any channel from here directly.. maybe by adding channel icon next to its related video or by an option in video menu.

if the swiping solution is not easy.. maybe a tab next to feed refresh bar would do the trick... or maybe make to tabs above the feed refresh bar, one called (new videos) and another called (channels).

also maybe you can make it an option to have it in settings... same as comments

**Comment by nhfo** [comment]
> > Wouldn't #3559 be a suitable solution?
> 
> It wouldn't solve the second problem I mentioned:
>
> > Another issue is that viewing and managing the list of subscriptions starts to become increasingly tedious by the time their number reaches the triple digits.
> 
> Plus, making an existing screen easier to manage would be better than introducing yet another tab on a Feed Page. It will reduce redundancy instead of increasing it.

@opusforlife2 does't your #3589 address this very issue? I think it is very neat.

> Actually, come to think of it, with this implemented, you could also easily do away with the channel group boxes above the subscriptions list:
> 
>     * As I said above, drop down arrows next to channel group headers could expand to show the contained subscriptions
> 
>     * The Channel Group headers _themselves_ could be made tappable, opening their Feeds.

In place expanding channel groups would be too cluttered IMHO.  Still how would it make it easy to manage group subscriptions?

IMO subscriptions page should get rid of old subscription listing completely, there is a "All" channel group and it is accessible from there once #3559 is implemented. Next channel groups box row should change to a grid or it should be a list (similar to playlists page).

**Comment by opusforlife2** [comment]
@nhfo Thanks! #3589 specifically addresses the problem I see with the Channel Group Subscription Picker only.

I thought of the channel group headers expanding here as mainly a way to visibly reduce the clutter of subscriptions in the Subscriptions tab. Instead of scrolling all the way down to an 'S', for example, you just expand the relevant header and you're there.

I don't think we can get rid of the subscription list completely, because there might be users who don't care about channel groups (there is already a humorously titled issue for it) and use the bare list. That is why I suggested a modification of the UI instead of a removal.

**Comment by nhfo** [comment]
> I thought of the channel group headers expanding here as mainly a way to visibly reduce the clutter of subscriptions in the Subscriptions tab. Instead of scrolling all the way down to an 'S', for example, you just expand the relevant header and you're there.

So did you mean that channel will be automatically grouped using first letter of channel name? And obviously these groups would have nothing to do with manually created channel groups?


> I don't think we can get rid of the subscription list completely, because there might be users who don't care about channel groups (there is already a humorously titled issue for it) and use the bare list. That is why I suggested a modification of the UI instead of a removal.

I have an idea, I suppose a somewhat related issue already exists. There should be another option for things that can be on main page (settings > content > content of main page), specifically channel group picker. When selected it will display a list of all channel groups (list will also contain "All" group), once a group is selected, user will select one of two options from a dialogue a) feed b) channels list.

With this if a user would want to have old style subscription list on main page, All group/channel list can be added to main page.

**Comment by opusforlife2** [comment]
@nhfo No, by relevant header, I definitely mean channel group header. The alphabet example is what we currently have to do: think of the first letter of the channel name and try to scroll directly to it, where we can get thrown off track by words like 'the'.

As for your second para, the issue already exists (https://github.com/TeamNewPipe/NewPipe/issues/3319), as does the pull request (https://github.com/TeamNewPipe/NewPipe/pull/3556), which has an apk you can test.

**Comment by SameenAhnaf** [comment]
I don't know why things have to be so complicated. Just add a gesture "Double tap on a group to see the grouped channels". Channels should be displayed in alphabetical order as usual.

**Comment by SameenAhnaf** [comment]
I don't know why things have to be so complicated. Just add a gesture "Double tap on a group to see the grouped channel list". Channels should be displayed in alphabetical order as usual.

Also, there should be 'Group' (suggested in #3588) 'Share' and 'Unsubscribe' tap for long press on those channels (just like in subscriptions tab). 

One more thing, while accessing a channel group list, '+2' icon on the right corner of title bar (list view) or on the bottom left corner of thumbnail (grid view) if the channel is duplicated on 2 more groups. I could have suggested this feature for subscription tab. But that would clutter a whole tap with hundreds of subscriptions.

**Comment by timur-g** [comment]
"Me too" would like channel groups to group, expand and collapse, thus reducing the number of subscriptions. Have a triangle facing down when expanded or facing right when collapsed, instead of the current star which I think has no special purpose.  
Note #12674 so good time to think this over.

**Comment by timur-g** [comment]
Here is the idea for dropdown groups (here it is to the right, or may be to the left icon) and subscription source,
Both are very needed with dozens or hundreds of subscriptions. 

<img width="368" height="973" alt="Image" src="https://github.com/user-attachments/assets/377f469f-4b37-495c-9af1-59852cc664d4" /> 

[newpipe-subs-mockup.html](https://github.com/user-attachments/files/30989809/newpipe-subs-mockup.html)
